Fix taxel centers and normals

rhaschke opened the pull request:
Centers and normals were off in L2 layout.

guihomework committed:
Fixed L1/L2/L3 taxels as well as common taxels

L2 and L1 taxels had wrong origin and scale/rot not applied
causing the pos/ori to be wrong. L3 pos/ort were already correct.
Additionally all taxels were cleaned to be only faces, not closed cuboids
that had too many vertices and normals, unwanted for the pos/ori computation

rhaschke commented on model/tactile_markers.urdf.xacro:
```diff
@@ -42,6 +42,10 @@
 			<parent link="${prefix}${finger}_middle_link"/>
 			<tactile channel="${channel}">
 				<xacro:taxel mesh="${F}mo" idx="${tactMap.get(F+'mo', -1)}"/>
+				<!--tmm does not exist for L1 thumb, but can be tmo which is larger and contains tmm -->
+				<xacro:if value="${layout=='L1' and finger=='thumb'}">
+					<xacro:taxel mesh="${F}mm" idx="${tactMap.get(F+'mo', -1)}"/>
+				</xacro:if>
 				<xacro:taxel mesh="${F}mm" idx="${tactMap.get(F+'mm', -1)}"/>
```
```suggestion
				<!--tmm does not exist for L1, but is covered by tmo as well -->
				<xacro:taxel mesh="${F}mm" idx="${tactMap.get(F+'mm', tactMap.get(F+'mo', -1) if layout=='L1' else -1)}"/>
```

guihomework replied:
none of the text are clear in fact. and I don't find how to do another suggestion or reject this one...

guihomework replied:
it took me some time to ensure this optimization of the code is doing the same due mentally needing to evaluate the code to understand what it does, while the if case is straight forward because it explicitly refers to the thumb. 

With the optimization we have to access an external document, to verify that there is no other ${F}mm that is not missing in another finger to be sure it works only for the thumb. For me, this opacifies  the understanding for future developments, and would for sure be missed in case one new layout has a pinky tmm missing or so. 

I confirm it should do the job for now, but don't like it.

guihomework commented:
@rhaschke This is not ready for a PR yet. We wanted to discuss the need or not to change also the taxel center with projection or not. Hence me not creating a PR

rhaschke commented:
Open issues:
- improve taxel location of U-shaped taxel shapes, e.g. projecting the current contact point onto the taxel surface (along the normal direction)
- more testing
- flipped normals to be verified

guihomework commented on model/tactile_markers.urdf.xacro:
```diff
@@ -42,6 +42,10 @@
 			<parent link="${prefix}${finger}_middle_link"/>
 			<tactile channel="${channel}">
 				<xacro:taxel mesh="${F}mo" idx="${tactMap.get(F+'mo', -1)}"/>
+				<!--tmm does not exist for L1 thumb, but can be tmo which is larger and contains tmm -->
```
```suggestion
				<!--tmm index does not exist for L1 thumb but tmo+tmm meshes cover the tmo L1 patch -->
```

guihomework committed:
Fixed more pos/ori issues on L2 and L3

Some rot/scale were still not applied on the L2right model
as well as 2 origins + rot/scale on L3right.
L2left had all normals flipped and outside recomputation
not active in the script.

guihomework commented:
Latest commit should fix the center and normals for all left/right L2/L3. See attached displays
![L2Left_center_normals](https://user-images.githubusercontent.com/6583181/95090113-320f9b00-0725-11eb-98be-c0e0cf037d71.png)
![L2Right_center_normals](https://user-images.githubusercontent.com/6583181/95090128-3471f500-0725-11eb-9f8d-f73cfc004825.png)
![L3Left_center_normals](https://user-images.githubusercontent.com/6583181/95090142-36d44f00-0725-11eb-9d36-e220ec9a608a.png)
![L3Right_center_normals](https://user-images.githubusercontent.com/6583181/95090145-389e1280-0725-11eb-869c-1bf1b55e7262.png)

rhaschke commented on model/tactile_markers.urdf.xacro:
```diff
@@ -42,6 +42,10 @@
 			<parent link="${prefix}${finger}_middle_link"/>
 			<tactile channel="${channel}">
 				<xacro:taxel mesh="${F}mo" idx="${tactMap.get(F+'mo', -1)}"/>
+				<!--tmm index does not exist for L1 thumb but tmo+tmm meshes cover the tmo L1 patch -->
+				<xacro:if value="${layout=='L1' and finger=='thumb'}">
+					<xacro:taxel mesh="${F}mm" idx="${tactMap.get(F+'mo', -1)}"/>
+				</xacro:if>
 				<xacro:taxel mesh="${F}mm" idx="${tactMap.get(F+'mm', -1)}"/>
```
@guihomework: What do you think about this one:
I want to avoid having mm listed twice. This way it becomes clear that mm is used but refers to two different indices.
```suggestion
				<!--tmm index does not exist for L1 thumb but tmo+tmm meshes cover the tmo L1 patch -->
				<xacro:taxel mesh="${F}mm" idx="${tactMap.get(F+'mo', -1) if (layout=='L1' and finger=='thumb') else tactMap.get(F+'mm', -1)}"/>
```

guihomework replied:
I prefer the version you suggest now, to clearly see the special case.

What about the other changes done in #5 , where other taxels are also listed twice (ptmp ptop both connected to ptop for instance). To be consistent with you wish to not have them twice, one would need to change it the same way.

I suggest to not merge this suggestion here, but fix (in a separate commit) the previous PR #5  first with these xacro changes and keep the taxel center fix separate.

rhaschke commented:
Otherwise looks good to me.
